### PR TITLE
Support watching app CRs in org namespace

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/20331

This PR adds support for watching app CRs in the organization namespace with the cluster label selector. This will be used for default-apps-openstack.

The changes are backwards compatible to watch app CRs in the cluster namespace to support existing clusters.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,14 +6,14 @@ require (
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/ghodss/yaml v1.0.0
 	github.com/giantswarm/apiextensions-application v0.3.0
-	github.com/giantswarm/app/v6 v6.3.0
+	github.com/giantswarm/app/v6 v6.3.1-0.20220118135250-241129fd3cd0
```
Comment (rossf7) [inline go.mod:9]: Needs https://github.com/giantswarm/app/pull/198

File: go.mod [hunk 2]
```diff
@@ -6,14 +6,14 @@ require (
...
 	github.com/giantswarm/appcatalog v0.6.0
 	github.com/giantswarm/apptest v1.0.0
 	github.com/giantswarm/backoff v1.0.0
 	github.com/giantswarm/errors v0.3.0
 	github.com/giantswarm/helmclient/v4 v4.9.0
 	github.com/giantswarm/k8sclient/v6 v6.1.0
-	github.com/giantswarm/k8smetadata v0.7.1
+	github.com/giantswarm/k8smetadata v0.7.2-0.20220118115255-6ec499e985ff
```
Comment (rossf7) [inline go.mod:16]: Needs Needs https://github.com/giantswarm/k8smetadata/pull/23

File: helm/app-operator/templates/configmap.yaml [hunk 3]
```diff
@@ -16,6 +16,8 @@ data:
     service:
       app:
         unique: {{ include "resource.app.unique" . }}
+        watchNamespace: '{{ .Values.app.watchNamespace }}'
+        workloadClusterID: '{{ .Values.app.workloadClusterID }}'
```
Comment (rossf7) [inline helm/app-operator/templates/configmap.yaml:20]: When watching app CRs in the org namespace these values will be set by cluster-apps-operator.

**Comment by rossf7** [comment]
Tested on gauss

File: service/controller/app/resource/chart/desired.go [hunk 4]
```diff
@@ -78,9 +81,23 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
...
-func generateAnnotations(input map[string]string, appNamespace string) map[string]string {
+func formatChartName(app v1alpha1.App, clusterID string) string {
+	// Chart CR name should match the app CR name when installed in the
+	// same cluster.
+	if key.InCluster(app) {
+		return app.Name
+	}
+
+	// If the app CR has the cluster ID as a prefix or suffix we remove it
+	// as its redundant in the remote cluster.
```
[the remark this answers is not in the document]
Comment (tfussell) [reply service/controller/app/resource/chart/desired.go]: Good call, this felt unnecessary when looking at the WC.

**Review by tfussell — APPROVED** [review]

**Pull request merged**